src/containers/header/search-form/search-bar/SearchBar.jsx

```diff
@@ -0,0 +1,30 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+import Input from '../../../../components/common/Input';
+
+import './search-bar.scss';
+
+const SearchBar = (props) => {
+    const { onSearchBarChange } = props;
+
+    console.log('rendering SearchBar ...');
+
+    return (
+        <Input
+            id="movie-search-bar"
+            className="search-bar"
+            name="movie"
+            type="text"
+            title="Find your movie"
+            onChangeHandler={ onSearchBarChange }
```

Comment:
This is because of lifting state up which I moved too high. The details are explained in one of the comments above regarding `SearchForm` dependency of `Header` state.